core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -291,10 +291,13 @@ public static Int128 valueOf(BigInteger value)
         catch (Exception e) {
             throw new ArithmeticException("Decimal overflow");
         }
-        throwIfOverflows(result.getHigh(), result.getLow());
+        if (overflows(result.getHigh(), result.getLow())) {
+            throw new ArithmeticException("Decimal overflow");
```

Comment:
should this also throw a TrinoException now? What about the exception caught on 291?

Reply:
yes, please throw here as well